### PR TITLE
Implement add to/remove from Favorites folder menu for a conversation

### PR DESCRIPTION
**Pull request by gizemb**
### Issues

Add "Add to Favorites" and "Remove from Favorites" menu items for a conversation & call related service methods.

#### APK
[Download build #133](http://10.10.124.11:8080/job/Pull%20Request%20Builder/133/artifact/build/artifact/wire-dev-PR2339-133.apk)
[Download build #147](http://10.10.124.11:8080/job/Pull%20Request%20Builder/147/artifact/build/artifact/wire-dev-PR2339-147.apk)
[Download build #148](http://10.10.124.11:8080/job/Pull%20Request%20Builder/148/artifact/build/artifact/wire-dev-PR2339-148.apk)
[Download build #149](http://10.10.124.11:8080/job/Pull%20Request%20Builder/149/artifact/build/artifact/wire-dev-PR2339-149.apk)
[Download build #150](http://10.10.124.11:8080/job/Pull%20Request%20Builder/150/artifact/build/artifact/wire-dev-PR2339-150.apk)
[Download build #152](http://10.10.124.11:8080/job/Pull%20Request%20Builder/152/artifact/build/artifact/wire-dev-PR2339-152.apk)
[Download build #153](http://10.10.124.11:8080/job/Pull%20Request%20Builder/153/artifact/build/artifact/wire-dev-PR2339-153.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala [hunk 1]
```diff
@@ -17,6 +17,7 @@
  */
 package com.waz.zclient.conversationlist
 
+import com.waz.log.LogSE._
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala]: Controllers are in UI, so his should be `import com.waz.zclient.log.LogUI._` (although I guess we could merge these two now that we merged projects).

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala [hunk 2]
```diff
@@ -93,18 +94,18 @@ class ConversationListController(implicit inj: Injector, ec: EventContext)
       members <- Signal.sequence(incomingConvs.map(c => membersStorage.activeMembers(c.id).map(_.find(_ != selfUserId))):_*)
     } yield (incomingConvs, members.flatten)
 
-  lazy val foldersWithConvs: Signal[Map[FolderId, Set[ConvId]]] = foldersService.flatMap(_.foldersWithConvs)
+  def foldersWithConvs: Signal[Map[FolderId, Set[ConvId]]] = foldersService.flatMap(_.foldersWithConvs)
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala]: Why the change from `lazy val` to `def`?

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala [hunk 3]
```diff
@@ -135,11 +136,14 @@ class ConversationListController(implicit inj: Injector, ec: EventContext)
     allIds <- allFolderIds
   } yield favId.fold(allIds)(allIds - _)
 
-  def addToFavourites(convId: ConvId): Future[Unit] = for {
+  def addToFavourites(convId: ConvId) = (for {
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala]: For public methods it's customary to specify the returned type.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/conversation/FoldersService.scala [hunk 4]
```diff
@@ -51,10 +50,13 @@ class FoldersServiceImpl(foldersStorage: FoldersStorage,
   private implicit val ev = EventContext.Global
 
   override def favouritesFolderId: Future[Option[FolderId]] =
-    foldersStorage.getByType(FolderData.FavouritesFolderType).map {
-      case head :: _ => Some(head.id)
-      case Nil       => None
-    }
+    foldersStorage.getByType(FolderData.FavouritesFolderType).map(i =>
+      if (i == Nil || i.isEmpty) {
```
Comment (makingthematrix) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/conversation/FoldersService.scala]: `Nil` and an empty list are the same things is Scala, and the previous version is more customary in Scala

Comment (gizemb) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/conversation/FoldersService.scala]: If I switch back to previous way, it somehow cannot calculate and I see an empty Conversation Menu.

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala (hunk 2 above)
Comment (gizemb) [reply app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala]: I thought that we shouldn't be caching this but since it's a Signal, we're already aware of any changes. Reverted back.

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala (hunk 3 above)
Comment (gizemb) [reply app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala]: done

**Comment by makingthematrix** [comment]
build again
